src/physics/utilities/equation_solver.hpp

```diff
@@ -171,9 +157,9 @@ class EquationSolver : public mfem::Solver {
   std::unique_ptr<mfem::Solver> prec_;
 
   /**
-   * @brief The linear solver object, either direct (SuperLU) or iterative
+   * @brief The linear solver object, either custom, direct (SuperLU), or iterative
    */
-  std::variant<std::unique_ptr<mfem::IterativeSolver>, std::unique_ptr<mfem::SuperLUSolver>> lin_solver_;
+  std::variant<std::unique_ptr<mfem::IterativeSolver>, std::unique_ptr<mfem::SuperLUSolver>, mfem::Solver*> lin_solver_;
```

Comment:
I think it goes beyond this PR, but will this `variant` get huge as we have various solver options?

Reply:
How many solver options does a production code normally have?

Reply:
if the custom solvers inherit from `mfem::Solver` or `mfem::IterativeSolver` it shouldn't need to grow, right?